Refactored code to be more functional

wesclemens opened the pull request:
Broke script up into several functions to make it easier to understand
also moved highlighting into actual error messages and check to see if
Git has highlighting enabled. Highlighting call will be chosen from Git
configuration now. Also check to see if PHP-CLI is installed

ReekenX commented:
Code looks great.

One comment (before accepting it) about code which contains:

<pre>
+        echo "PHP-cli not installed"
+        exit 0;
</pre>

It's error, but exit code is `success`. This is a mistake or with some purpose?

wesclemens commented:
It was on purpose. If it errors out it would prevent you from committing. I didn't want to break git on systems that didn't have the php cli installed. Although, a better error message maybe warranted.

wesclemens commented:
After using today I found a bug where the on disk version of the file is check instead of the staged version of the file. I have added a patch for this to the pull request. I have also added a silent mode as I don't like it when it prints messages when there are no errors.
